Makefile

```diff
@@ -455,7 +455,7 @@ generate-go-conversions-docker-infrastructure: $(CONVERSION_GEN) ## Generate con
 generate-go-openapi: $(OPENAPI_GEN) $(CONTROLLER_GEN) ## Generate openapi go code for runtime SDK
 	@mkdir -p ./tmp/sigs.k8s.io; ln -s $(ROOT_DIR) ./tmp/sigs.k8s.io/; cd ./tmp; \
 	for pkg in "api/v1beta1" "$(EXP_DIR)/runtime/hooks/api/v1alpha1"; do \
-		$(MAKE) clean-generated-openapi-definitions SRC_DIRS="./$${pkg}"; \
+		$(MAKE) clean-generated-openapi-definitions -C sigs.k8s.io/cluster-api SRC_DIRS="./$${pkg}"; \
```

Comment:
I'm not sure if this works for everyone. We had issues with this generate target before where it behaves different depending on if CAPI is cloned in it's "GOPATH" location or outside of it.

I would suggest to move this change out of this PR if possible

Reply:
I had made a clarification as to why I made this change [here](https://github.com/kubernetes-sigs/cluster-api/pull/8289#discussion_r1137624172), but I resolved it by mistake so it probably didn't show up.

I agree being inside/outside of GOPATH would change the way Go commands work, but I'm not sure if it would affect this clean target since it's just running `rm`. Either way I made this change because I got an error when running the `make generate` command that Killian had suggested, so I'm not too attached to this change.

Reply:
Ah nope my bad. Change looks fine. I rememberd that we have some branching based on GOPATH but it is not in this target. So this should be just a straightforward fix